### PR TITLE
Implement missing Split-Path parameters

### PR DESCRIPTION
**Pull request by mrward**
Support the following parameters:

   * -Leaf
   * -Qualifier
   * -NoQualifier
   * -Resolve
   * -IsAbsolute
   * -LiteralPath

Also support wildcards:

    Split-Path *.txt -Leaf -Resolve

### CODE REVIEW

File: Source/Microsoft.PowerShell.Commands.Management/SplitPathCommand.cs [hunk 1]
```diff
@@ -1,34 +1,175 @@
...
+using System.Collections.Generic;
 using System.Collections.ObjectModel;
 using System.Management;
 using System.Management.Automation;
+using System.Management.Automation.Provider;
 
 namespace Microsoft.PowerShell.Commands
 {
-    [Cmdlet("Split", "Path")]
+    [Cmdlet(VerbsCommon.Split, "Path", DefaultParameterSetName = "ParentSet")]
```
Comment (JayBazuzi) [inline Source/Microsoft.PowerShell.Commands.Management/SplitPathCommand.cs]: Nice

**Comment by mrward** [comment]
Is this pull request OK to be merged into master?

**Comment by JayBazuzi** [comment]
Not sure how I missed that, sorry!

**Comment by staxmanade** [comment]
Hey @mrward 

On my Mac I get the following two errors but don't before this was merged?

```
		Errors and Failures:
		1) Test Failure : ReferenceTests.Commands.SplitPathTests.ResolveParentForTwoFilesUsingLiteralPaths
		     Expected string length 98 but was 114. Strings differ at index 1.
		  Expected: "/var/folders/rc/3bw9hqm90jj9wlgnz5bqf0gr0000gn/T\n/var/folders..."
		  But was:  "/private/var/folders/rc/3bw9hqm90jj9wlgnz5bqf0gr0000gn/T\n/pri..."
		  ------------^
		at ReferenceTests.Commands.SplitPathTests.ResolveParentForTwoFilesUsingLiteralPaths () [0x00000] in <filename unknown>:0
		2) Test Failure : ReferenceTests.Commands.SplitPathTests.ResolveParentForTwoFilesUsingWildcard
		     Expected string length 98 but was 114. Strings differ at index 1.
		  Expected: "/var/folders/rc/3bw9hqm90jj9wlgnz5bqf0gr0000gn/T\n/var/folders..."
		  But was:  "/private/var/folders/rc/3bw9hqm90jj9wlgnz5bqf0gr0000gn/T\n/pri..."
		  ------------^
		at ReferenceTests.Commands.SplitPathTests.ResolveParentForTwoFilesUsingWildcard () [0x00000] in <filename unknown>:0
```

Here is my mono version info (if this means anything)...

```
Mono JIT compiler version 3.6.0 (tarball Mon Sep 15 14:40:37 PDT 2014)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
	TLS:           normal
	SIGSEGV:       altstack
	Notification:  kqueue
	Architecture:  amd64
	Disabled:      none
	Misc:          softdebug 
	LLVM:          supported, not enabled.
	GC:            sgen
```

**Comment by mrward** [comment]
I can reproduce this with Mono 3.12 on the Mac.

The problem seems to be that /var on the Mac is a symlink for /private/var and the file APIs return /private/var when resolving the full path but Path.GetTempPath () returns /var

Testing this with a console app:

	string tempDir = Path.GetTempPath ();
	string fileName = Path.Combine (tempDir, "test.txt");
	File.WriteAllText (fileName, "Test");

	Console.WriteLine ("FileName: {0}", fileName);
	Console.WriteLine ("Path.GetFullPath(FileName): {0}", Path.GetFullPath (fileName));

	Directory.SetCurrentDirectory (tempDir);
	Console.WriteLine ("Path.GetFullPath(\"test.txt\"): {0}", Path.GetFullPath ("test.txt"));
	Console.WriteLine ("CurrentDirectory: {0}", Directory.GetCurrentDirectory ());

On the Mac the output is:

    FileName: /var/folders/ky/yjlcxlc52mb04njk1z7k9qfw0000gn/T/test.txt
    Path.GetFullPath(FileName): /var/folders/ky/yjlcxlc52mb04njk1z7k9qfw0000gn/T/test.txt
    Path.GetFullPath("test.txt"): /private/var/folders/ky/yjlcxlc52mb04njk1z7k9qfw0000gn/T/test.txt
    CurrentDirectory: /private/var/folders/ky/yjlcxlc52mb04njk1z7k9qfw0000gn/T

Looking at some of the Mono tests they [prefix the path with /private](https://github.com/mono/mono/blob/mono-3.12.0-branch/mcs/class/corlib/Test/System.IO/FileStreamTest.cs#L120-L121) if they are on the Mac when checking the paths match.

Another way to fix this may be to change the test so it changes the working directory to the temp directory and calls Directory.GetCurrentDirectory () to find the full path and then this path is used in the test Assert instead of the path returned from Path.GetTempPath ().

**Comment by JayBazuzi** [comment]
I'm happy with a quick fix at this point, since we're already broken, then see about CI on Mac, then look at more elegant fixes.

**Pull request merged**